### PR TITLE
ft: S3C-3118-flatten-objmd-retentioninfo

### PR DESCRIPTION
**Pull request by dora-korpar**
Flatten the retention info stored in object md.

### CODE REVIEW

File: lib/models/ObjectMD.js [hunk 1]
```diff
@@ -902,25 +902,39 @@ class ObjectMD {
...
+     */
+    setRetentionDate(date) {
+        this._data.retentionDate = date;
         return this;
     }
 
     /**
-     * Returns object retention information
+     * Returns object retention mode
+     * @return {object} metadata object
```
Comment (jonathan-gramain) [inline lib/models/ObjectMD.js]: You need to update the returned type

File: lib/models/ObjectMD.js [hunk 2]
```diff
@@ -902,25 +902,39 @@ class ObjectMD {
...
+     */
+    getRetentionMode() {
+        return this._data.retentionMode;
+    }
+
+    /**
+     * Returns object retention retain until date
      * @return {object} metadata object
```
Comment (jonathan-gramain) [inline lib/models/ObjectMD.js]: same

**Review by jonathan-gramain — APPROVED** [review]

**Review by ilkescality — APPROVED** [review]
LGTM; so glad to see this just work seamlessly without any further adjustment in anywhere else 🎉

**Comment by dora-korpar** [comment]
> LGTM; so glad to see this just work seamlessly without any further adjustment in anywhere else 🎉

wait till you see cloudserver 😆

**Comment by dora-korpar** [comment]
@bert-e reset

**Comment by ilkescality** [comment]
just realized that this isn't :) for a sec I actually thought this _is_ cloudserver, no wonder why it felt too good to be true 😂

**Comment by dora-korpar** [comment]
@bert-e approve

**Pull request merged**